src/chainstate/stacks/boot/bns.clar

```diff
@@ -106,18 +106,11 @@
     (asserts!
       (> namespace-len u0)
       (err ERR_NAMESPACE_BLANK))
-    (ok (get value (fold 
-      element-at 
-      NAMESPACE_PRICE_TIERS 
-      { limit: (min u8 namespace-len), cursor: u0, value: u0 })))))
-
-(define-private (element-at (i uint) (acc (tuple (limit uint) (cursor uint) (value uint))))
-  (if (is-eq (get cursor acc) (get limit acc))
-    { limit: (get limit acc), cursor: (+ u1 (get cursor acc)), value: i }
-    { limit: (get limit acc), cursor: (+ u1 (get cursor acc)), value: (get value acc) }))
-  
+    (ok (unwrap-panic
+      (element-at NAMESPACE_PRICE_TIERS (min u7 (- namespace-len u1)))))))
+
 (define-private (get-exp-at-index (buckets (list 16 uint)) (index uint))
-  (get value (fold element-at buckets { limit: index, cursor: u0, value: u0 })))
+  (unwrap-panic (element-at buckets index)))
```

Comment:
@lgalabru -- do these updates to `bns.clar` look correct to you?

I think there was an off-by-one error in the indexing of the namespace price tiers (index 0 in the tiers should be the price of a length 1 namespace, not a length 0 namespace). I'd guess this affects the tests in `bns-test`, but they all fail because they don't run against the git repo's version of the clarity-bin:

```
  ExecutionError: Launch contract failed with bad exit code 1: Contract initialization error: 
    UnknownFunction("element-at")
    Near:
    [SymbolicExpression { expr: Atom(ClarityName("let")), id: 421, span: Span { start_line: 105, start_column: 4, end_line: 105, end_column: 6 } }, SymbolicExpression { expr: List([SymbolicExpression { expr: List([SymbolicExpression { expr: Atom(ClarityName("namespace-len")), id: 424, span: Span { start_line: 105, start_column: 10, end_line: 105, end_column: 22 } }, SymbolicExpression { expr: List([SymbolicExpression { expr: Atom(ClarityName("len")), id: 426, span: Span { start_line: 105, start_column: 25, end_line: 105, end_column: 27 } }, SymbolicExpression { expr: Atom(ClarityName("namespace")), id: 427, span: Span { start_line: 105, start_column: 29, end_line: 105, end_column: 37 } }]), id: 425, span: Span { start_line: 105, start_column: 24, end_line: 105, end_column: 38 } }]), id: 423, span: Span { start_line: 105, start_column: 9, end_line: 105, end_column: 39 } }]), id: 422, span: Span { start_line: 105, start_column: 8, end_line: 105, end_column: 40 } }, SymbolicExpression { expr: List([SymbolicExpression { expr: Atom(ClarityName("asserts!")), id: 429, span: Span { start_line: 106, start_column: 6, end_line: 106, end_column: 13 } }, SymbolicExpression { expr: List([SymbolicExpression { expr: Atom(ClarityName(">")), id: 431, span: Span { start_line: 107, start_column: 8, end_line: 107, end_column: 8 } }, SymbolicExpression { expr: Atom(ClarityName("namespace-len")), id: 432, span: Span { start_line: 107, start_column: 10, end_line: 107, end_column: 22 } }, SymbolicExpression { expr: LiteralValue(UInt(0)), id: 433, span: Span { start_line: 107, start_column: 24, end_line: 107, end_column: 24 } }]), id: 430, span: Span { start_line: 107, start_column: 7, end_line: 107, end_column: 26 } }, SymbolicExpression { expr: List([SymbolicExpression { expr: Atom(ClarityName("err")), id: 435, span: Span { start_line: 108, start_column: 8, end_line: 108, end_column: 10 } }, SymbolicExpression { expr: Atom(ClarityName("ERR_NAMESPACE_BLANK")), id: 436, span: Span { start_line: 108, start_column: 12, end_line: 108, end_column: 30 } }]), id: 434, span: Span { start_line: 108, start_column: 7, end_line: 108, end_column: 31 } }]), id: 428, span: Span { start_line: 106, start_column: 5, end_line: 108, end_column: 32 } }, SymbolicExpression { expr: List([SymbolicExpression { expr: Atom(ClarityName("ok")), id: 438, span: Span { start_line: 109, start_column: 6, end_line: 109, end_column: 7 } }, SymbolicExpression { expr: List([SymbolicExpression { expr: Atom(ClarityName("unwrap-panic")), id: 440, span: Span { start_line: 109, start_column: 10, end_line: 109, end_column: 21 } }, SymbolicExpression { expr: List([SymbolicExpression { expr: Atom(ClarityName("element-at")), id: 442, span: Span { start_line: 110, start_column: 8, end_line: 110, end_column: 17 } }, SymbolicExpression { expr: Atom(ClarityName("NAMESPACE_PRICE_TIERS")), id: 443, span: Span { start_line: 110, start_column: 19, end_line: 110, end_column: 39 } }, SymbolicExpression { expr: List([SymbolicExpression { expr: Atom(ClarityName("min")), id: 445, span: Span { start_line: 110, start_column: 42, end_line: 110, end_column: 44 } }, SymbolicExpression { expr: LiteralValue(UInt(7)), id: 446, span: Span { start_line: 110, start_column: 46, end_line: 110, end_column: 46 } }, SymbolicExpression { expr: List([SymbolicExpression { expr: Atom(ClarityName("-")), id: 448, span: Span { start_line: 110, start_column: 50, end_line: 110, end_column: 50 } }, SymbolicExpression { expr: Atom(ClarityName("namespace-len")), id: 449, span: Span { start_line: 110, start_column: 52, end_line: 110, end_column: 64 } }, SymbolicExpression { expr: LiteralValue(UInt(1)), id: 450, span: Span { start_line: 110, start_column: 66, end_line: 110, end_column: 66 } }]), id: 447, span: Span { start_line: 110, start_column: 49, end_line: 110, end_column: 68 } }]), id: 444, span: Span { start_line: 110, start_column: 41, end_line: 110, end_column: 69 } }]), id: 441, span: Span { start_line: 110, start_column: 7, end_line: 110, end_column: 70 } }]), id: 439, span: Span { start_line: 109, start_column: 9, end_line: 110, end_column: 71 } }]), id: 437, span: Span { start_line: 109, start_column: 5, end_line: 110, end_column: 72 } }]

      68 |     provider = await NativeClarityBinProvider.create(allocations, dbFileName, binFile);
```


Or should I just rename `element-at` to my like `bns-element-at` in this PR to avoid this issue for now?

Reply:
Yeah working with BNS became particularly tedious yesterday, after merging the new clarity features, breaking backward compatibility. 
We might have a solution for this soon with https://github.com/blockstack/clarity-js-sdk/pull/83 thanks to @friedger.

Reply:
I'm taking note of the off by one error and will investigate thank you!